src/controllers/sessionController.ts

```diff
@@ -5,21 +5,45 @@
 import * as express from "express";
 import {
   IResponseErrorInternal,
+  IResponseErrorNotFound,
   IResponseErrorValidation,
   IResponseSuccessJson,
   ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseErrorValidation,
   ResponseSuccessJson
 } from "italia-ts-commons/lib/responses";
 
 import { PublicSession } from "../../generated/backend/PublicSession";
 import { SessionsList } from "../../generated/backend/SessionsList";
 
-import { isLeft } from "fp-ts/lib/Either";
+import { sequenceT } from "fp-ts/lib/Apply";
+import { isLeft, toError } from "fp-ts/lib/Either";
+import { identity } from "fp-ts/lib/function";
+import { fromEither, taskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { SuccessResponse } from "src/types/commons";
 import RedisSessionStorage from "../services/redisSessionStorage";
+import RedisUserMetadataStorage from "../services/redisUserMetadataStorage";
 import { withUserFromRequest } from "../types/user";
 
+type ResponseLockUserSessionT =
+  | IResponseErrorInternal
+  | IResponseErrorValidation
+  | IResponseErrorNotFound
+  | IResponseSuccessJson<SuccessResponse>;
+
+type ResponseUnockUserSessionT =
+  | IResponseErrorInternal
+  | IResponseErrorValidation
+  | IResponseSuccessJson<SuccessResponse>;
+
 export default class SessionController {
```

Comment:
I'd rather create a new controller since this one is dedicated to user's api (is uses `withUserFromRequest`) while the new methods here are being called internally and have a dedicated API spec